### ISSUE TITLE
pyodbc don't find and open exist driver

### ISSUE THREAD

**Issue by Varvara-Gornostaeva** [mkleehammer/pyodbc#539]
### Environment

- Python: 3.6.8 (pyenv enviroment)
- pyodbc: 4.0.26
- OS: Ubuntu 18.04.2 LTS
- DB: MSSQL 2016 (remote connection)
- driver: "ODBC Driver 17 for SQL Server" 

### Issue
pyodbc.Error: ('01000', "[01000] [unixODBC][Driver Manager]Can't open lib '/opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1' : file not found (0) (SQLDriverConnect)")

I reinstalled the driver several times by
- Instruction: https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-2017
- Just install deb package: https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-2017

❯ odbcinst -j                                                                                                                                                                                           [12:55:26]
unixODBC 2.3.7
DRIVERS............: /etc/odbcinst.ini
SYSTEM DATA SOURCES: /etc/odbc.ini
FILE DATA SOURCES..: /etc/ODBCDataSources
USER DATA SOURCES..: /home/varvara/.odbc.ini
SQLULEN Size.......: 8
SQLLEN Size........: 8
SQLSETPOSIROW Size.: 8

❯ odbcinst -q -d -n "ODBC Driver 17 for SQL Server"                                                                                                                                                     [12:55:28]
[ODBC Driver 17 for SQL Server]
Description=Microsoft ODBC Driver 17 for SQL Server
Driver=/opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1
UsageCount=1

I tried just open file
❯ /opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1                                                                                                                                             [12:55:33]
[1]    18755 segmentation fault (core dumped)  /opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1

Please help me understand the problem.

I investigated all existing error reports (StackOverflow, Microsoft drivers GitHub, askUbuntu, etc), re-checked docs pyodbc  - don't resolve problem
And method from 
https://github.com/mkleehammer/pyodbc/issues/240
Not work fo me

Thanks! c:

**Comment by v-chojas** [comment]
Does `isql` work? Try this:

`isql -v -k 'Driver=ODBC Driver 17 for SQL Server;server=<your server>;UID=<your user>;PWD=<your pass>'`

If that does not work and fails with the same error, then `strace` it to find out which file it cannot open.
If that works, but pyODBC does not, then you will need to `strace` your Python script for the same purpose.

**Comment by v-makouz** [comment]
Also can you do `ldd /opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1` just to make sure nothing suspicious is going on there

**Comment by Varvara-Gornostaeva** [comment]
@v-makouz 
❯ ldd /opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1      [11:46:18]
	linux-vdso.so.1 (0x00007fff2b1bc000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f8067eba000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f8067cb2000)
	libodbcinst.so.2 => /usr/lib/x86_64-linux-gnu/libodbcinst.so.2 (0x00007f8067a97000)
	libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f8067654000)
	libkrb5.so.3 => /usr/lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007f806737e000)
	libgssapi_krb5.so.2 => /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f8067133000)
	libssl.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f8066ecb000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007f8066cc4000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f806693b000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f806659d000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f8066385000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f8066166000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8065d75000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f80684c3000)
	libk5crypto.so.3 => /usr/lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007f8065b43000)
	libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007f806593f000)
	libkrb5support.so.0 => /usr/lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f8065734000)
	libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007f8065530000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f8065315000)
 ✔ varvara ⮀ ~ ⮀
❯ ldd /opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1 | grep 'not found'                                                              [16:36:07]
 ✘ varvara ⮀ ~ ⮀
❯        

I apologize for not writing right away. Checked this option

**Comment by Varvara-Gornostaeva** [comment]
@v-makouz 
Your command is work.
https://github.com/mkleehammer/pyodbc/wiki/Module#connect
But this same credentials do not work for pyodbc.connect :c

![zhiza](https://user-images.githubusercontent.com/24761400/54118987-ddac1880-4426-11e9-9ce4-12fa07a3d50a.png)

**Comment by keitherskine** [comment]
@Varvara-Gornostaeva In the image you included in your last message, the username and password to your database are visible in the error message at the bottom of the image.  You should probably change that password.

**Comment by v-chojas** [comment]
Since `isql` works but Python doesn't, `strace` your Python process and find which of the files it cannot open, and why.

**Comment by Varvara-Gornostaeva** [comment]
@keitherskine Yes, I noticed it late (very stupid). Changed creds and deleted messages

**Comment by Varvara-Gornostaeva** [comment]
Sorry guys, I found the problem 
I upgrade the python version from 3.6.3 to 3.6.8 -> interaction with the driver stopped workingApparently, this is a problem either in `pyodbc` (wait for the update) or in ubuntu or in pyenv.
Or the reason is in the chaos of dependencies

Thank you all for the answers!
 I am closing the issue for now, as it is easier to roll back to the old version.

**Comment by sorenvind** [comment]
I just stumbled into this bug and spent some hours investigating what was happening. The reason was that we built on top of docker container `python:3.7-slim`, and that was updated recently to be based on debian buster instead of debian stretch. The binary msodbcsql17 driver requires `libssl1.0.2` which is not available in debian buster out of the box. 

For others hitting this bug (and using docker): The simple fix is to switch your dockerfile to be based on `python:3.7-slim-stretch`.
